extensions/typescript/src/features/taskProvider.ts

```diff
@@ -28,6 +28,7 @@ const exists = (file: string): Promise<boolean> =>
 
 interface TypeScriptTaskDefinition extends vscode.TaskDefinition {
 	tsconfig: string;
+	label: string;
```

Comment:
Adding label to the identifier and making it mandatory will break all existing typescript task customizations already inserted into a tasks.json. So we have to make sure we make this backwards compatible. I see the following possibilities:

- have a complete different type: 'typescript-watch'. However this might be confusing
- add an optional `option` property and add the 'watch' in case of a watching task. However this requires that I address: https://github.com/Microsoft/vscode/issues/31970 which is targeted for September
- only generate watch or non watch tasks based on a setting. 

Having both watch and non watch tasks in there might be overwhelming. May be we should make watch the default and have a setting to switch to non watch. @egamma you once opted for both. Is this something you still think is necessary?

Reply:
Thanks I'll give this a try

We previously had the logic for using the `compileOnSave` option to enable watch over regular build. When @kieferrm and I discussed this further though, I think we wanted to show both tasks. This is not overwhelming if you only have a handful of tsconfigs or jsconfigs but does not look great for a project like vscode

Reply:
Which option will you give a try?

Reply:
I like the `option` property one. Do any other builtin tasks providers run in watch mode by default though?

Reply:
@mjbvz  I addressed #31970. Let me know if you see problems.